elastic/eui
# [Emotion] Convert EuiDataGrid `gridStyle`s to Emotion (Part 3)

## Summary

Part of #6394

This PR looks intimidating in terms of file diff, but I promise the majority of it is just Storybooks! As such I strongly recommend [following along by commit](https://github.com/elastic/eui/pull/8006/commits). The TL;DR of changes:

- Converts styles impacted by `gridStyle` to Emotion (rows, borders, and cell density)
- Addresses a long-standing TODO in EuiDataGrid where we were using static numbers to calculate row height to inherit dynamically from our EUI theme tokens (637a2ba)
- Fixes border style permutations that look broken on prod, mainly around underline/overline header & footers (https://github.com/elastic/eui/pull/8006/commits/30c6f9bccec1ccc8bf7e6209030a2880c8df05de?w=1)
- A ton of Storybook cleanup (last 4-5 commits):
  - I wanted a way to get the nested objects in Storybook's controls table, so I created a separate `stories.utils.tsx` file as a way to generate fake 'components' that would fool Storybook's react docgen, as well as to share a stateful datagrid:
  <img width="1500" alt="" src="https://github.com/user-attachments/assets/a53066e0-7949-4f77-b5a7-6dbfaaa73fd7">
  This lets us easily reuse the same dataset and columns etc across multiple stories and story subdirectories. And IMO, EuiDataGrid is well complex enough to warrant multiple `.stories` files 🫠 

## QA

- https://eui.elastic.co/pr_8006/storybook/?path=/story/tabular-content-euidatagrid-grid-styles--playground - test each permutation via the controls table and confirm no broken visuals
- https://eui.elastic.co/pr_8006/storybook/?path=/story/tabular-content-euidatagrid-row-heights - quick smoke check for no broken visuals in this folder and that the controls make sense
- https://eui.elastic.co/pr_8006/#/tabular-content/data-grid-schema-columns#footer-row

### General checklist

- Browser QA
    ~- [ ] Checked in **mobile**~
    ~- [ ] Checked for **accessibility** including keyboard-only and screenreader modes~
- Docs site QA
    - [x] Props have proper **autodocs** (using `@default` if default values are missing)
    ~- [ ] Checked **[Code Sandbox](https://codesandbox.io/)** works for any docs examples~
- Code quality checklist
- Release checklist
    ~- [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)~
- Designer checklist - N/A
